### DISCUSSION TITLE
Limit switch state for home assistant

### DISCUSSION THREAD

**Discussion by baskarz** [arendst/Tasmota#18414] [Q&A]
Hi all, I'm making a garage door controller and have successfully got the relay to control the door to work correctly, no false triggering of relay on boot and pulse the relay to activate garage door.
Now I have installed 2 limit switches at the open and closed positions and would like to show the states of these switches in home assistant. 
I'm using mqtt in home assistant for all my other tasmota devices and works fantastic. 
I've searched around with no definite answer. 
What would be the best approach to this?
 
Thanks

**Comment by barbudor** [comment]
I have such a setting on Tasmota but not using HA so I can't help you with the HA part.
The switches are using Switchmode 16 so they are detached from the relay and will send dedicated messages when changing state
In addition Tasmota is sending the switches status on the teleperiod SENSOR message. 

I'm not using HA but Node-red where I can directly process the MQTT messages. 

It may be (but I can't be certain) that Tasmota only report in it's discovery messages about relays, lights and sensors. So you may have to deal with manual configuration. 
I remember seeing people talking about 'binary sensors' so that might give you some way of searching here and in HA forums.

**Reply by baskarz** [reply]
Thanks!
I think that might be just what I'm looking for.
HA should auto detect the switch instances once created.
I shall try it later.

**Reply by baskarz** [reply]
Ok so setting switchmode(x) 16 works in that I can see state change in console.
I just need to work out how Configure sending mqqt messages to HA.

Thanks for pointing me in the right direction. I'll report back once I work it out for anyone else in the future.

**Reply by barbudor** [reply]
the message should already been send to MQTT
You should see them in the console
So it's more about configuring HA to catch them up and display as you want
I also have a VL53L1X which tells me if there is a car in there

Node-RED dash board:
![image](https://user-images.githubusercontent.com/2996042/232309782-b027ffb8-3554-4223-b841-0f65baa05121.png)
